moveit_simple/src/robot.cpp

```diff
@@ -0,0 +1,1606 @@
+  : Robot(nh, robot_description, group_name)
+{
+  try
+  {
+    ik_base_frame_ = ik_base_frame;
+    ik_tip_frame_ = ik_tip_frame;
+    this->computeIKSolverTransforms();
+  }
+  catch (tf2::TransformException& ex)
+  {
+    ROS_ERROR_STREAM("Failed to compute transforms between the base/tip frames defined"
+                     << " in the SRDF and the base/tip frames defined for the IK solver");
+    throw IKSolverTransformException("Failed to compute transforms between the base/tip"
+                                     " frame defined in the SRDF and the base/tip frames defined for the IK solver");
+  }
+}
+
+Robot::~Robot()
+{
+}
+
+void Robot::refreshRobot()
+{
+  ROS_INFO_STREAM("Loading MoveIt objects based on, robot description: " << robot_description_
+                                                                         << ", group name: " << planning_group_);
+
+  std::lock_guard<std::recursive_mutex> guard(m_);
+  robot_model_loader_.reset(new robot_model_loader::RobotModelLoader(robot_description_));
+  robot_model_ptr_ = robot_model_loader_->getModel();
+
+  virtual_robot_state_.reset(new moveit::core::RobotState(robot_model_ptr_));
+  virtual_robot_state_->setToDefaultValues();
+
+  planning_scene_.reset(new planning_scene::PlanningScene(robot_model_ptr_));
+
+  joint_group_ = robot_model_ptr_->getJointModelGroup(planning_group_);
+
+  ROS_INFO_STREAM("Calculating all positions assuming, root: " << robot_model_ptr_->getRootLinkName() << ", and tool: "
+                                                               << robot_model_ptr_->getLinkModelNames());
+  ROS_INFO_STREAM("MoveIt object loaded");
+
+  virtual_visual_tools_.reset(new moveit_visual_tools::MoveItVisualTools(
+      robot_model_ptr_->getRootLinkName(), nh_.getNamespace() + "/rviz_visual_tools", robot_model_ptr_));
+  virtual_visual_tools_->loadRobotStatePub(nh_.getNamespace() + "/display_robot_state");
+  virtual_visual_tools_->publishRobotState(virtual_robot_state_, rviz_visual_tools::PURPLE);
+  virtual_visual_tools_->trigger();
+}
+
+void Robot::addTrajPoint(const std::string& traj_name, const Eigen::Affine3d pose, const std::string& frame,
+                         double time, const InterpolationType& type, const unsigned int num_steps,
+                         const std::string& point_name)
+{
+  std::lock_guard<std::recursive_mutex> guard(m_);
+
+  ROS_INFO_STREAM("Attempting to add " << point_name << " to " << traj_name << "relative to" << frame << " at time "
+                                       << time);
+
+  try
+  {
+    Eigen::Affine3d pose_rel_robot = transformToBase(pose, frame);
+    std::unique_ptr<TrajectoryPoint> point =
+        std::unique_ptr<TrajectoryPoint>(new CartTrajectoryPoint(pose_rel_robot, time, point_name));
+    addTrajPoint(traj_name, point, type, num_steps);
+  }
+  catch (tf2::TransformException& ex)
+  {
+    ROS_ERROR_STREAM("Add to trajectory failed for arbitrary pose point: " << ex.what());
```

Comment:
ROS_ERROR_STREAM isn't the place for this output. As users run the tests, it pollutes the console and misleads developers into thinking things have gone wrong when they haven't. If this is being handled upstream I would reconsider printing to ROS_ERROR_STREAM